src/measure_theory/function/conditional_expectation/basic.lean

```diff
@@ -2142,6 +2157,13 @@ begin
     ((condexp_L1_mono hf hg hfg).trans_eq (condexp_ae_eq_condexp_L1 hm _).symm),
 end
 
+lemma condexp_nonneg {f : α → ℝ} (hf : 0 ≤ᵐ[μ] f) (hfint : integrable f μ) :
```

Comment:
You don't need `hfint`: if the function is not integrable, then the conditional expectation is zero and the inequality is verified.

Also, could you generalize this lemma from the reals to some type verifying the same hypotheses as in `condexp_mono`?

And finally, while you are adding this, could you also add a `nonpos` version?